src/WaveTrack.cpp

```diff
@@ -1255,201 +1258,189 @@ void WaveTrack::SyncLockAdjust(double oldT1, double newT1)
+        // a new clip.
+        return;
+
+    //wxPrintf("Check if we need to make room for the pasted data\n");
+
+    bool editClipCanMove = GetEditClipsCanMove();
+
+    // Make room for the pasted data
+    if (editClipCanMove) {
+        if (!singleClipMode) {
             // We need to insert multiple clips, so split the current clip and
             // move everything to the right, then try to paste again
             if (!IsEmpty(t0, GetEndTime())) {
-               auto tmp = Cut(t0, GetEndTime()+1.0/mRate);
-               Paste(t0 + insertDuration, tmp.get());
+                auto tmp = Cut(t0, GetEndTime() + 1.0 / mRate);
+                Paste(t0 + insertDuration, tmp.get());
             }
-         }
-         else {
+        }
+        else {
             // We only need to insert one single clip, so just move all clips
             // to the right of the paste point out of the way
-            for (const auto &clip : mClips)
+            for (const auto& clip : mClips)
             {
-               if (clip->GetStartTime() > t0-(1.0/mRate))
-                  clip->Offset(insertDuration);
+                if (clip->GetPlayStartTime() > t0 - (1.0 / mRate))
+                    clip->Offset(insertDuration);
             }
-         }
-      }
+        }
+    }
 
-      if (singleClipMode)
-      {
-         // Single clip mode
-         // wxPrintf("paste: checking for single clip mode!\n");
+    if (singleClipMode)
+    {
+        // Single clip mode
+        // wxPrintf("paste: checking for single clip mode!\n");
 
-         WaveClip *insideClip = NULL;
+        WaveClip* insideClip = nullptr;
 
-         for (const auto &clip : mClips)
-         {
+        for (const auto& clip : mClips)
+        {
             if (editClipCanMove)
             {
-               if (clip->WithinClip(t0))
-               {
-                  //wxPrintf("t0=%.6f: inside clip is %.6f ... %.6f\n",
-                  //       t0, clip->GetStartTime(), clip->GetEndTime());
-                  insideClip = clip.get();
-                  break;
-               }
+                if (clip->WithinPlayRegion(t0))
+                {
+                    //wxPrintf("t0=%.6f: inside clip is %.6f ... %.6f\n",
+                    //       t0, clip->GetStartTime(), clip->GetEndTime());
+                    insideClip = clip.get();
+                    break;
+                }
             }
             else
             {
-               // If clips are immovable we also allow prepending to clips
-               if (clip->WithinClip(t0) ||
-                     TimeToLongSamples(t0) == clip->GetStartSample())
-               {
-                  insideClip = clip.get();
-                  break;
-               }
+                // If clips are immovable we also allow prepending to clips
+                if (clip->WithinPlayRegion(t0) ||
+                    TimeToLongSamples(t0) == clip->GetSequenceStartSample())
```

Comment:
Should this be GetPlayStartSample()?

Reply:
Hm, yes, and this makes possible the case with prepending data in `Paste`. Thanks!